src/libos/src/net/socket/unix/stream/stream.rs

```diff
@@ -68,8 +69,24 @@ impl Stream {
         return_errno!(ENOTCONN, "the socket is not connected");
     }
 
-    // TODO: create the corresponding file in the fs
-    pub fn bind(&self, addr: &Addr) -> Result<()> {
+    pub fn bind(&self, addr: &mut Addr) -> Result<()> {
+        let inode_num = if let Addr::File(path, inode) = addr {
```

Comment:
Rewrite it with clear logic
```
if let Addr::File(path, inode_num) = addr {
        // create the corresponding file in the fs and fill Addr with its inode
        let corresponding_inode_num = {
            let current = current!();
            let fs = current.fs().read().unwrap();
            let file_ref = fs.open_file(&addr.path_str(), CreationFlags::O_CREAT.bits(), 0o777)?;
            file_ref.metadata()?.inode
        };
        *inode_num = Some(corresponding_inode_num);
    }
```

Reply:
Then why not clone the `addr.path_str()` to a String and pass the reference of the newly cloned String to `open_file`?

Reply:
Hi, I think the code below can work

```
if let Addr::File(inode_num, unix_path) = addr {
            // create the corresponding file in the fs and fill Addr with its inode
            let corresponding_inode_num = {
                let current = current!();
                let fs = current.fs().read().unwrap();
                let file_ref =
                    fs.open_file(&unix_path.path_str(), CreationFlags::O_CREAT.bits(), 0o777)?;
                file_ref.metadata()?.inode
            };
            *inode_num = Some(corresponding_inode_num);
        }
```

Reply:
This logic is more clear. Changed a little to make it work. Thanks for your review.

Reply:
Yes, mine is the same. Please take a look. Thank you!